error_test.go

```diff
@@ -45,3 +46,42 @@ func ExampleValue() {
 	// B2 nil error:7
 	// B2 with error:[#2 EOF value must be nil]
 }
+
+func ExampleValue2() {
+	fmt.Print("Value2 nil error: ")
+	fmt.Println(must.Value2(1, 2, nil))
+
+	fmt.Println("Panic message:", must.Panic(func() {
+		must.Value2(1, 2, errors.New("P"))
+	}).(error).Error())
```

Comment:
I think we should write one good example for all Value# functions and write the rest as tests. What do you think?

Reply:
I think we should merge to have good coverage, and then work on doc improvements.

Doc and examples for `Value3`/`V3`/`Value4`/`V4` are quite useless because redundant (if someone has read `Value2`/`V2` that should be enough), but we could improve examples by using concrete functions from stdlib.

Reply:
The coding style (error type assertion) is fine as a test, but should not be in an example because it's not guaranteed by the API <https://github.com/xiegeo/must#compatibility>

> Values used in panics are only for debugging. No guarantees are provided on how panics are constructed.

If you have a use case. It's possible to provide a function or change the docs to reflect it. My thinking on this is if you need panic to be created in a particular way, it's better to roll your own than use a dependency.

Reply:
Well the response to the second question is in https://github.com/xiegeo/must#compatibility

Reply:
Would it be ok if I transform the `Example`s into just `Test`s?

Reply:
Yes, write as a test is ok.